crypto/bio/bss_mem.c

```diff
@@ -319,19 +320,33 @@ static long mem_ctrl(BIO *b, int cmd, long num, void *ptr)
     case BIO_CTRL_SET_CLOSE:
         b->shutdown = (int)num;
         break;
+    case BIO_CTRL_DGRAM_QUERY_MTU:
+    case BIO_CTRL_DGRAM_GET_MTU_OVERHEAD:
     case BIO_CTRL_WPENDING:
+    case BIO_CTRL_GET_KTLS_SEND:
+    case BIO_CTRL_GET_KTLS_RECV:
         ret = 0L;
         break;
     case BIO_CTRL_PENDING:
         ret = (long)bm->length;
         break;
+    case BIO_CTRL_DGRAM_SET_NEXT_TIMEOUT:
+    case BIO_CTRL_PUSH:
+    case BIO_CTRL_POP:
     case BIO_CTRL_DUP:
     case BIO_CTRL_FLUSH:
-        ret = 1;
+    case BIO_C_SET_NBIO:
+    case BIO_CTRL_DGRAM_SET_MTU:
+    ret = 1;
         break;
-    case BIO_CTRL_PUSH:
-    case BIO_CTRL_POP:
+
+    case BIO_CTRL_SET_PREFIX:
+    case BIO_CTRL_SET_INDENT:
+    case BIO_CTRL_GET_INDENT:
     default:
+        ERR_raise(ERR_LIB_BIO, ERR_R_UNSUPPORTED);
+        snprintf(buf, sizeof(buf), "%d", cmd);
+        ERR_add_error_data(2, "cmd=", buf);
```

Comment:
I think this should be reverted and no error code should be raised for unsupported commands.

Reply:
Sigh, this BIO cmd handling has been done culpably sloppily - some return codes are plainly wrong, and the default case often mixes integer `0` results and Boolean 'false` results.
I've removed my attempts to clean this can of worms from this PR, and for now it is in quarantine in draft PR  #18850.